### PR TITLE
feat(detection): add TrainingSlicer for tiled training data generation

### LINKED ISSUE #1996: Add support for TrainingSlicer
### Search before asking

- [x] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Description

I am happy to give this a shot, but wanted to make sure this would be useful.

For small object detection, I think it would be useful to slice training images. It would need a slightly different interface to InferenceSlicer.

Something like:
```
class TrainingSlicer:

    def __init__(
        self,
        callback: Callable[[np.ndarray, Detections], list[tuple[np.ndarray, Detections]]],
        slice_wh: tuple[int, int] = (320, 320),
        ....
    ):

    def __call__(self, image: np.ndarray, detections: Detections) -> list[tuple[np.ndarray, Detections]]:
        pass
```

### Use case

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by daniel-devlab**
Adds TrainingSlicer, the training-time counterpart to InferenceSlicer (#1996). Slices ground-truth Detections into tiles for training small-object models.

<details>
<summary>Before submitting</summary>

- [x] Self-reviewed the code
- [x] Updated documentation, follow [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods)
- [x] Added docs entry for autogeneration (if new functions/classes)
- [x] Added/updated tests
- [x] All tests pass locally

</details>

## Description

Adds `TrainingSlicer`, the training-time counterpart to `InferenceSlicer`. Instead of running a model callback on each tile and merging predictions back, it slices an image's existing ground-truth `Detections` to match a grid of tiles, translating box, oriented-box, and mask coordinates into each tile's local frame. This is the standard way to turn a dataset of large images into fixed-size training crops so small objects occupy a larger fraction of what the detector sees during training (the same technique used by tools like SAHI).

## Type of Change

- ✨ New feature (non-breaking change which adds functionality)

## Motivation and Context

Closes #1996. For small-object detection/segmentation, training on full-resolution images means small objects are a tiny fraction of the input, which hurts training signal. Slicing the training set into tiles (as `InferenceSlicer` already lets you do at inference time) is a standard mitigation, but `supervision` had no equivalent for ground-truth annotations. This fills that gap.

## Changes Made

- Added `TrainingSlicer` (`src/supervision/detection/tools/training_slicer.py`): slices an image and its `Detections` into a grid of tiles, with `slice_wh`, `overlap_wh`, `min_visibility` (drops annotations a tile boundary cuts below a configurable fraction of their original area), and `drop_empty_slices` (optionally omit background-only tiles).
- Reused `move_detections` from `inference_slicer.py` for box/OBB/mask coordinate translation, so `CompactMask` and dense masks are both handled without duplicating that logic.
- Exported `TrainingSlicer` from `supervision/__init__.py`.
- Added `docs/detection/tools/training_slicer.md` and registered it in `mkdocs.yml`.
- Added unit tests in `tests/detection/tools/test_training_slicer.py` covering the tiling grid, input validation, box localization/clipping at tile boundaries, the `min_visibility` threshold, mask cropping, and `drop_empty_slices`.

## Testing

- [x] I have tested this code locally
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] All new and existing tests pass

Ran the new test file (16 tests, all passing) plus the full existing test suite to confirm no regressions, `ruff check`/`ruff format`, and `mypy` on the new module.

## Google Colab (optional)

Colab link: not yet added - happy to add one on request if it'd help review.

## Screenshots/Videos (optional)

N/A (no visual/annotator changes)

## Additional Notes

One deliberate deviation from the interface sketched in #1996: that proposal had both a constructor `callback` and a `__call__` that already returns the sliced result directly, which overlap awkwardly. I dropped the `callback` param — `__call__` returns `list[tuple[image, Detections]]` directly, which is simpler and lets the caller do whatever they want downstream (write files, feed an augmentation pipeline, etc.), consistent with how `InferenceSlicer` and other tools in this package are composed. Open to feedback if maintainers prefer the callback-based shape instead.

Also worth noting: OBB coordinates are translated into each tile's local frame but not clipped to the tile rectangle (clipping a rotated box against an axis-aligned rectangle produces a >4-vertex polygon), so an OBB straddling a tile boundary may have vertices outside that tile. Documented as a known limitation in the class docstring.

### CODE REVIEW

**Comment by Borda** [comment]
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/roboflow/supervision?pullRequest=2518) before we can accept your contribution.

@daniel-devlab ^^ 🙏

**Comment by Borda** [comment]
@daniel-devlab, it looks like this was a one-time, fire-and-forget PR with little follow-up on reviews, and without a signed CLA we cannot do anything to get it to land, so we're closing it for now... However, if you want to continue, we are happy to reopen it and see it land... 🦝
